lib/kms/in_memory/backend.js

```diff
@@ -0,0 +1,81 @@
+import common from '../common';
+
+export const kms = [];
+let count = 1;
+
+export const backend = {
+    /*
+     * Target implementation will be async in some aspects. let mimic it
+     */
+
+    /*
+     * cb(err, masterKeyId: string)
+     */
+    createBucketKey: function createBucketKeyMem(bucketName, log, cb) {
+        process.nextTick(() => {
+            kms[count] = common.createDataKey();
```

Comment:
either comment that using the object data key for convenience or just create random bytes